x-pack/plugins/stack_connectors/server/connector_types/lib/servicenow/service.test.ts

```diff
@@ -805,6 +876,220 @@ describe('ServiceNow service', () => {
     });
   });
 
+  describe('closeIncident', () => {
+    const closeIncidentMock = () =>
+      Promise.resolve({
+        title: 'INC01',
+        id: '1',
+        pushedDate: '2020-03-10T12:24:20.000Z',
+        url: 'https://example.com/nav_to.do?uri=incident.do?sys_id=1',
+      });
+
+    // new connectors
+    describe('import set table', () => {
+      test('it closes the incident correctly with incident id', async () => {
+        const spyOnCloseIncident = jest.spyOn(service, 'closeIncident');
+        spyOnCloseIncident.mockImplementation(closeIncidentMock);
```

Comment:
Why do we mock the `closeIncident` function and not `requestMock`?

Reply:
I tried `requestMock ` but it is not working as expected because `closeIncident` and `getIncident` has different response (one with result as array and one with result as object) which throw error.

Reply:
You can mock `requestMock ` as you like, with multiple responses for different calls, and only in your test (or your `describe` block).